### PR TITLE
Nmap os detection

### PR DESCRIPTION
**Pull request by joesizzlin**
### Description

Closes [Term Entry] Nmap: OS Detection #1313   

Wrote an entry for [Nmap ](https://www.codecademy.com/resources/docs/cybersecurity/nmap)about the OS Detection option.  
Located under docs/content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md

Includes:  
- A description of what the option does.
- A listing of the option itself: -O
- A syntax section showing the option in use.
- An example showing Nmap's output with the option. (Use scanme.nmap.org as a target.)

### Type of Change

- Adding a new entry

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by joesizzlin** [comment]
This is my first public contribution. Thanks

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:18]:
```suggestion
One of Nmap's best known features is remote **OS detection** using TCP/IP stack fingerprinting. With OS detection, Nmap sends a series of TCP and UDP packets to the specified host and examines the responses. Nmap then compares the results to its *nmap-os-db* database of more than 2,600 known OS fingerprints and prints out the OS details if there is a match.  
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:21]:
```suggestion
> **Note:** Root privileges are required to properly perform the scan.
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:3]:
```suggestion
Description: 'One of the best-known features of Nmap is remote OS detection using TCP/IP stack fingerprinting.'
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:24]:
```suggestion
OS detection is enabled with the `-O` option.
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:26]: Delete this and include it in the line above, as I did with my suggestion.

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:22]: Eliminate extra line break.

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:37]: Eliminate extra line break

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:44]:
```suggestion
The above conducts an Nmap scan with OS detection on `<target>`.
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:28]:
```suggestion
OS detection is controlled with the following options (be sure to include with `-O`):
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:29]:
```suggestion
- Limit OS detection to promising targets (`--osscan-limit`)
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:30]:
```suggestion
    - This can save substantial time by only scanning targets that have a criteria that makes OS detection more effective, which is if there is at least one open and one closed TCP port are found. Nmap will not attempt OS detection against hosts which do not meet that criteria.
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:31]:
```suggestion
- Guess OS detection results (`--osscan-guess; --fuzzy`)
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:32]:
```suggestion
    - When Nmap is unable to detect a perfect OS match, it sometimes offers up near-matches as possibilities. The match has to be very close for Nmap to do this by default. Either of these (equivalent) options make Nmap guess more aggressively. Nmap will still tell when an imperfect match is printed and display its confidence level (percentage) for each guess.
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:33]:
```suggestion
- Set the maximum number of OS detection tries against a target (`--max-os-tries`)
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:35]:
```suggestion
- Increase verbosity level to OS detection Scan (`-v`)
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (SSwiniarski) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:48]:
```suggestion
The following example runs a Nmap Scan and enables OS detection on the site [scanme.nmap.org](http://scanme.nmap.org/):
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@joesizzlin I did a first review and left some comments. I rewrote your description because the Metadata doesn't like apostrophes in the description.

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:22]: removed extra line break

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:33]: Removed extra line break

**Review by joesizzlin — COMMENTED** [review]
Suggested changes from first review have been addressed.

**Review by joesizzlin — COMMENTED** [review]
Suggested changes from review have been addressed

**Review by SSwiniarski — APPROVED** [review]
@joesizzlin, looks good for a second review.

**Comment by SSwiniarski** [comment]
@joesizzlin, Not yet, another maintainer will do a second review and may have more edits before we merge.

**Comment by Dusch4593** [comment]
Hi @joesizzlin 👋🏻 I'll take a look for review 2️⃣

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:3]:
```suggestion
Description: 'Remotely scans a target host and presents details of its operating system if there is a match.'
```

For term entries in Docs, let's set up the `Description` such that it begins with a present-tense verb 😄

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@joesizzlin Left you some comments/suggestions. 😄

**Comment by joesizzlin** [comment]
@Dusch4593 thanks for reviewing! I will chip away those suggestions over the next several days :smiley:

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md]: added suggested fingerprint info below ```shell snippet

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (joesizzlin) [reply content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:17]: I thought about this one for a few minutes, I eventually warmed up to the idea.
Lets run with it :smiley:

**Review by joesizzlin — COMMENTED** [review]
When I told you "several days" before, I actually meant an hour or two :wink: 
Suggestions looked good and are committed.

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:75]:
```suggestion

Each fingerprint includes a freeform textual description of the OS, and a classification which provides the following:
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:37]:
```suggestion
- `--osscan-limit`: Limits detection to only host targets with at least one open and one closed TCP port, and ignores hosts that do not meet this criteria.
- `--osscan-guess; --fuzzy`: Presents possible results if a perfect OS match can't be found.
  - The match has to be very close for Nmap to do this by default. Either of these (equivalent) options make Nmap guess more aggressively. Nmap will still tell when an imperfect match is printed and display its confidence level (0% - 100%) for each guess.
- `--max-os-tries`: Sets the maximum number of detection attempts against a target.
  - By default, Nmap tries five times if conditions are favorable or two times when conditions aren't so good. A specific amount of tries can be specified; the lower the amount, the quicker the scan.
- `-v`: Provides more information from an OS detection scan, including an incremental ID for each IP packet header sent.
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:16]:
```suggestion
**Operating system (OS) detection** is a feature in Nmap that remotely [scans](https://www.codecademy.com/resources/docs/cybersecurity/nmap/nmap-scan) a target host and presents details of its operating system if there is a match. TCP/IP stack fingerprinting is used to send a series of probes (e.g., TCP and UDP packets) to the specified host and examines the responses. The results are then compared to the `nmap-os-db` database 2,600+ known OS fingerprints, and the details are printed if there is a match.
```

File: content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md
Comment (Dusch4593) [inline content/cybersecurity/concepts/nmap/terms/os-detection/os-detection.md:21]:
```suggestion

> **Note:** Root privileges are required to properly perform the scan.

```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@joesizzlin Looking great! Just a few more comments/suggestions 😄

**Review by joesizzlin — COMMENTED** [review]
@Dusch4593  Additional suggestions reviewed and committed :smiley:

**Review by Dusch4593 — APPROVED** [review]
@joesizzlin LGTM! 👍🏻

**Comment by Dusch4593** [comment]
Congrats and thank you for contributing to Codecademy Docs, @joesizzlin! 🎊 🎉 🥳

The entry should now be live: https://www.codecademy.com/resources/docs/cybersecurity/nmap/os-detection

Screen Shot 2023-01-13 at 9 46 49 PM

Looking forward to your next Docs contribution! 😄

**Comment by joesizzlin** [comment]
@Dusch4593 thank you for your help :tada:

**Pull request merged**
